File: ios/Classes/Reminder.swift
import Foundation
import CoreLocation


struct Reminder {
  var id: String = ""
  var lat: Double = 0
  var lng: Double = 0
  var radius: Int = 0
  var name: String = ""
  var active: Bool = false
  var message: String = ""
  var snoozeTill: Double = 0
  var snoozeLength: Int = 0

  init(fromDictionary r: [String: Any]) {
    id = r["id"] as! String
    lat = r["lat"] as! Double
    lng = r["lng"] as! Double
    radius = r["radius"] as! Int
    name = r[name] as! String
    active = r["active"] != nil ? r["active"] as! Bool : false
    message = r["message"] as! String
    snoozeTill = r["snoozeTill"] as! Double
    snoozeLength = r["snoozeLength"] as! Int
  }

  func toDict() -> [String: Any] {
        var dict = [String: Any]()
        dict["id"] = id
        dict["lat"] = lat
        dict["lng"] = lng
        dict["radius"] = radius
        dict["name"] = name
        dict["active"] = active
        dict["message"] = message
        dict["snoozeTill"] = snoozeTill
        dict["snoozeLength"] = snoozeLength

        return dict
    }

    func getLocation2D() -> CLLocationCoordinate2D {
        return CLLocationCoordinate2D(latitude: lat, longitude: lng)
    }
    
    func getLocation() -> CLLocation {
        return CLLocation(latitude: lat, longitude: lng)
    }

}

File: android/src/main/kotlin/com/brandonlehr/geofencing/Reminder.kt
package com.brandonlehr.geofencing

class Reminder (
    var _id: String = "",
    var _lat: Double = 0.0,
    var _lng: Double = 0.0,
    var _radius: Int = 0,
    var _name: String = "",
    var _active: Boolean = false,
    var _message: String = "",
    var _snoozeTill: Double = 0.0,
    var _snoozeLength: Int = 0
    ) {
    var id: String = _id
    var lat: Double = _lat
    var lng: Double = _lng
    var radius: Int = _radius
    var name: String = _name
    var active: Boolean = _active
    var message: String = _message
    var snoozeTill: Double = _snoozeTill
    var snoozeLength: Int = _snoozeLength

    constructor(args: Map<String, Any>) : this() {
        this.id = args["id"] as String
        this.lat = args["lat"] as Double
        this.lng = args["lng"] as Double
        this.radius = args["radius"] as Int
        this.name = args["name"] as String
        this.active = args["active"] as Boolean
        this.message = args["message"] as String
        this.snoozeTill = args["snoozeTill"] as Double
        this.snoozeLength = args["snoozeLength"] as Int
    }

    fun toMap(): Map<String, Any> {
        val map = mutableMapOf<String, Any>()
        map["id"] = this.id
        map["lat"] = this.lat
        map["lng"] = this.lng
        map["radius"] = this.radius
        map["name"] = this.name
        map["active"] = this.active
        map["message"] = this.message
        map["snoozeTill"] = this.snoozeTill
        map["snoozeLength"] = this.snoozeLength
        return map
    }


}